core/src/impl/Kokkos_StringManipulation.hpp

```diff
@@ -173,7 +173,7 @@ KOKKOS_FUNCTION constexpr to_chars_result to_chars_i(char *first, char *last,
       unsigned_val = Unsigned(~value) + Unsigned(1);
     }
   }
-  unsigned int const len = to_chars_len(unsigned_val);
+  std::ptrdiff_t const len = to_chars_len(unsigned_val);
```

Comment:
Why was this change necessary?

Reply:
https://github.com/masterleinad/kokkos/actions/runs/4430586886/jobs/7772515059:
```
In file included from /__w/kokkos/kokkos/core/unit_test/TestStringManipulation.cpp:17:
/__w/kokkos/kokkos/core/src/impl/Kokkos_StringManipulation.hpp: In instantiation of 'constexpr Kokkos::Impl::to_chars_result Kokkos::Impl::to_chars_i(char*, char*, Integral) [with Integral = int]':
/__w/kokkos/kokkos/core/unit_test/TestStringManipulation.cpp:153:21:   required from 'constexpr bool {anonymous}::to_chars_helper(Integral, const char*) [with Integral = int]'
/__w/kokkos/kokkos/core/unit_test/TestStringManipulation.cpp:158:32:   required from here
/__w/kokkos/kokkos/core/src/impl/Kokkos_StringManipulation.hpp:177:20: error: comparison of integer expressions of different signedness: 'int' and 'const unsigned int' [-Werror=sign-compare]
  177 |   if (last - first < len) {
      |       ~~~~~~~~~~~~~^~~~~
/__w/kokkos/kokkos/core/src/impl/Kokkos_StringManipulation.hpp: In instantiation of 'constexpr Kokkos::Impl::to_chars_result Kokkos::Impl::to_chars_i(char*, char*, Integral) [with Integral = unsigned int]':
/__w/kokkos/kokkos/core/unit_test/TestStringManipulation.cpp:153:21:   required from 'constexpr bool {anonymous}::to_chars_helper(Integral, const char*) [with Integral = unsigned int]'
/__w/kokkos/kokkos/core/unit_test/TestStringManipulation.cpp:164:32:   required from here
/__w/kokkos/kokkos/core/src/impl/Kokkos_StringManipulation.hpp:177:20: error: comparison of integer expressions of different signedness: 'int' and 'const unsigned int' [-Werror=sign-compare]
/__w/kokkos/kokkos/core/src/impl/Kokkos_StringManipulation.hpp: In instantiation of 'constexpr Kokkos::Impl::to_chars_result Kokkos::Impl::to_chars_i(char*, char*, Integral) [with Integral = long long int]':
/__w/kokkos/kokkos/core/unit_test/TestStringManipulation.cpp:153:21:   required from 'constexpr bool {anonymous}::to_chars_helper(Integral, const char*) [with Integral = long long int]'
/__w/kokkos/kokkos/core/unit_test/TestStringManipulation.cpp:168:32:   required from here
/__w/kokkos/kokkos/core/src/impl/Kokkos_StringManipulation.hpp:177:20: error: comparison of integer expressions of different signedness: 'int' and 'const unsigned int' [-Werror=sign-compare]
/__w/kokkos/kokkos/core/src/impl/Kokkos_StringManipulation.hpp: In instantiation of 'constexpr Kokkos::Impl::to_chars_result Kokkos::Impl::to_chars_i(char*, char*, Integral) [with Integral = long long unsigned int]':
/__w/kokkos/kokkos/core/unit_test/TestStringManipulation.cpp:153:21:   required from 'constexpr bool {anonymous}::to_chars_helper(Integral, const char*) [with Integral = long long unsigned int]'
/__w/kokkos/kokkos/core/unit_test/TestStringManipulation.cpp:172:32:   required from here
/__w/kokkos/kokkos/core/src/impl/Kokkos_StringManipulation.hpp:177:20: error: comparison of integer expressions of different signedness: 'int' and 'const unsigned int' [-Werror=sign-compare]
```